### PR TITLE
added steps to check items count inside a folder

### PR DESCRIPTION
**Pull request by sushmita56**
added a step to check no. of items inside the provided folder

### CODE REVIEW

File: src/stepDefinitions/filesContext.js [hunk 1]
```diff
@@ -163,110 +169,141 @@ const assertContentOfLocalFileIsNot = function(fullPathOfLocalFile, actualConten
...
+    let actualCount = jsonResponse.elements[0].elements.length
+    actualCount = actualCount - 1
+    folder_count = parseInt(folder_count)
+    return assert.strictEqual(
+      actualCount,
+      folder_count,
+      `Test Failed...Expected ${folder_count} folders but got ${actualCount}`
+    )
+  }
+)
```
Comment (kiranparajuli589) [inline src/stepDefinitions/filesContext.js:309]: hmm...implementations are ditto.
you can avoid duplications like:
```suggestion
Then(
  'as user {string} folder {string} should contain {string} items',
  async function (userName, folderName, itemsCount) {
    const davPath = webdavHelper.createDavPath(userName, folderName)
    const resourcesInsideFolder = await webdavHelper.propfind(davPath, userName, [])
    let jsonResponse = convert.xml2json(resourcesInsideFolder)
    jsonResponse = JSON.parse(jsonResponse)
    let actualCount = jsonResponse.elements[0].elements.length
    actualCount = actualCount - 1
    return assert.strictEqual(
      actualCount,
      parseInt(itemsCount),
      `Expected ${folder_count} items inside ${folderName} but found ${actualCount}`
    )
  }
)
```

**Review by kiranparajuli589 — CHANGES_REQUESTED** [review]
also please revert all other changes which are not related

File: src/stepDefinitions/filesContext.js [hunk 2]
```diff
@@ -1,13 +1,16 @@
 const assert = require('assert')
 const fs = require('fs')
+var util = require('util')
```
Comment (Talank) [inline src/stepDefinitions/filesContext.js:3]:
```suggestion
const util = require('util')
```

File: src/stepDefinitions/filesContext.js [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
 const { Given, Then, After } = require('../context')
 const backendHelper = require('../helpers/backendHelper')
 const { move } = require('../helpers/webdavHelper')
 const path = require('../helpers/path')
 const { download } = require('../helpers/webdavHelper')
 const { client } = require('../config.js')
-const webdavHelper = require("../helpers/webdavHelper")
+const webdavHelper = require('../helpers/webdavHelper')
+
+var convert = require('xml-js')
```
Comment (Talank) [inline src/stepDefinitions/filesContext.js:13]:
```suggestion
const convert = require('xml-js')
```

**Review by Talank — COMMENTED** [review]
use const instead of var while loading modules.

**Review by saw-jan — CHANGES_REQUESTED** [review]
@sushmita56 you can undo these format changes for now and only add what you need in this PR.
We need to fix those unformatted files though.

File: src/stepDefinitions/filesContext.js [hunk 4]
```diff
@@ -9,6 +9,7 @@ const { download } = require('../helpers/webdavHelper')
 const { client } = require('../config.js')
 const webdavHelper = require("../helpers/webdavHelper")
 
+const convert = require('xml-js')
 const createdFiles = []
```
Comment (saw-jan) [inline src/stepDefinitions/filesContext.js:13]: just to group
```suggestion
const webdavHelper = require("../helpers/webdavHelper")
const convert = require('xml-js')

const createdFiles = []
```

File: src/stepDefinitions/filesContext.js [hunk 5]
```diff
@@ -270,3 +271,23 @@ Given(/^user "([^"]*)" has created folder "(.+)"$/, function(userId, folderName)
...
+    userName,
+    folderName,
+    itemsCount
+) {
+  const davPath = webdavHelper.createDavPath(userName, folderName)
+  const resourcesInsideFolder = await webdavHelper.propfind(davPath, userName, [])
+  let jsonResponse = convert.xml2json(resourcesInsideFolder)
+  jsonResponse = JSON.parse(jsonResponse)
+  let actualCount = jsonResponse.elements[0].elements.length
+  actualCount = actualCount - 1
```
Comment (saw-jan) [inline src/stepDefinitions/filesContext.js:286]: `xml-js` comes with `xml2js` function so you can directly convert XML to js object.
```suggestion
  const xmlResponse = await webdavHelper.propfind(davPath, userName, [])
  const resObj = convert.xml2js(xmlResponse)
  // also add some comment here why we need to subtract by 1
  const actualCount = resObj.elements[0].elements.length - 1
```

File: src/stepDefinitions/filesContext.js [hunk 6]
```diff
@@ -270,3 +271,23 @@ Given(/^user "([^"]*)" has created folder "(.+)"$/, function(userId, folderName)
...
+  itemsCount = parseInt(itemsCount)
+  return assert.strictEqual(
+    actualCount,
+    itemsCount,
+    `Test Failed...Expected ${itemsCount} folders but got ${actualCount}`
```
Comment (saw-jan) [inline src/stepDefinitions/filesContext.js:291]:
```suggestion
    `Expected '${itemsCount}' items inside the folder '${folderName}' but got '${actualCount}'`
```

**Review by saw-jan — CHANGES_REQUESTED** [review]
Some changes request

File: src/stepDefinitions/filesContext.js [hunk 7]
```diff
@@ -270,3 +269,22 @@ Given(/^user "([^"]*)" has created folder "(.+)"$/, function(userId, folderName)
...
+  'as user {string} folder {string} should contain {string} items', async function (
+    userName,
+    folderName,
+    itemsCount
+) {
+  const davPath = webdavHelper.createDavPath(userName, folderName)
+  const resourcesInsideFolder = await webdavHelper.propfind(davPath, userName, [])
+  const jsonResponse = convert.xml2js(resourcesInsideFolder)
+  let actualCount = jsonResponse.elements[0].elements.length
+  actualCount = actualCount - 1
```
Comment (kiranparajuli589) [inline src/stepDefinitions/filesContext.js:283]: why subtract? please add a comment about this.

File: src/stepDefinitions/filesContext.js [hunk 8]
```diff
@@ -270,3 +269,22 @@ Given(/^user "([^"]*)" has created folder "(.+)"$/, function(userId, folderName)
...
+Then(
+  'as user {string} folder {string} should contain {string} items', async function (
+    userName,
+    folderName,
+    itemsCount
+) {
+  const davPath = webdavHelper.createDavPath(userName, folderName)
+  const resourcesInsideFolder = await webdavHelper.propfind(davPath, userName, [])
+  const jsonResponse = convert.xml2js(resourcesInsideFolder)
+  let actualCount = jsonResponse.elements[0].elements.length
```
Comment (Talank) [inline src/stepDefinitions/filesContext.js:282]: `jsonResponse.elements[0].elements.length` does not give much of the information to the reader. Can you add a comment on this line as well? Which element that you are targeting?

**Review by saw-jan — APPROVED** [review]
LGTM

**Review by kiranparajuli589 — APPROVED** [review]

File: src/stepDefinitions/filesContext.js [hunk 9]
```diff
@@ -270,3 +271,21 @@ Given(/^user "([^"]*)" has created folder "(.+)"$/, function(userId, folderName)
...
 })
+
+Then(
+  'as user {string} folder {string} should contain {string} items',
+  async function (userName, folderName, itemsCount) {
+    const davPath = webdavHelper.createDavPath(userName, folderName)
+    const xmlResponse = await webdavHelper.propfind(davPath, userName, [])
+    const resObj = convert.xml2js(xmlResponse, { compact: true })
+    // 'd:response' contains entry for requested folder too
+    // so we need to subtract 1 to get the acutal items count
```
Comment (phil-davis) [inline src/stepDefinitions/filesContext.js:282]: Typo fixed in PR #102

**Pull request merged**